FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -0,0 +1,291 @@
+  _isTesterSignedIn = self.authState ? YES : NO;
+  return self;
+}
+
++ (void)load {
+  [FIRApp registerInternalLibrary:(Class<FIRLibrary>)self
+                         withName:@"firebase-appdistribution"
+                      withVersion:@"0.0.0"];  // TODO: Get version from podspec
+}
+
++ (NSArray<FIRComponent *> *)componentsToRegister {
+  FIRComponentCreationBlock creationBlock =
+      ^id _Nullable(FIRComponentContainer *container, BOOL *isCacheable) {
+    if (!container.app.isDefaultApp) {
+      NSLog(@"App Distribution must be used with the default Firebase app.");
+      return nil;
+    }
+
+    *isCacheable = YES;
+
+    return [[FIRAppDistribution alloc] initWithApp:container.app
+                                           appInfo:NSBundle.mainBundle.infoDictionary];
+  };
+
+  FIRComponent *component =
+      [FIRComponent componentWithProtocol:@protocol(FIRAppDistributionInstanceProvider)
+                      instantiationTiming:FIRInstantiationTimingEagerInDefaultApp
+                             dependencies:@[]
+                            creationBlock:creationBlock];
+  return @[ component ];
+}
+
++ (instancetype)appDistribution {
+  // The container will return the same instance since isCacheable is set
+
+  FIRApp *defaultApp = [FIRApp defaultApp];  // Missing configure will be logged here.
+
+  // Get the instance from the `FIRApp`'s container. This will create a new instance the
+  // first time it is called, and since `isCacheable` is set in the component creation
+  // block, it will return the existing instance on subsequent calls.
+  id<FIRAppDistributionInstanceProvider> instance =
+      FIR_COMPONENT(FIRAppDistributionInstanceProvider, defaultApp.container);
+
+  // In the component creation block, we return an instance of `FIRAppDistribution`. Cast it and
+  // return it.
+  return (FIRAppDistribution *)instance;
+}
+
+- (void)signInTesterWithCompletion:(FIRAppDistributionSignInTesterCompletion)completion {
+  NSURL *issuer = [NSURL URLWithString:@"https://accounts.google.com"];
+
+  [OIDAuthorizationService
+      discoverServiceConfigurationForIssuer:issuer
+                                 completion:^(OIDServiceConfiguration *_Nullable configuration,
+                                              NSError *_Nullable error) {
+                                   if (!configuration) {
+                                     NSLog(@"Error retrieving discovery document: %@",
+                                           [error localizedDescription]);
+                                     return;
+                                   }
+
+                                   NSString *redirectUrl = [@"dev.firebase.appdistribution."
+                                       stringByAppendingString:
+                                           [[[NSBundle mainBundle] bundleIdentifier]
+                                               stringByAppendingString:@":/launch"]];
```

Comment:
If you use stringByAppendingPathComponent, you don't have to define the "/":
https://developer.apple.com/documentation/foundation/nsstring/1417069-stringbyappendingpathcomponent?language=objc

Reply:
Done